qualtran/_infra/data_types.py

```diff
@@ -911,7 +911,7 @@ class QGF(QDType):
     characteristic: SymbolicInt
     degree: SymbolicInt
     irreducible_poly: Optional['galois.Poly'] = attrs.field()
-    element_repr: Literal["int", "poly", "power"] = attrs.field(default='int')
+    element_repr: Literal["int", "poly", "power"] = attrs.field(default='int', eq=False)
```

Comment:
Please add a test to `data_types_test.py` with examples of which `QGF` types should be equal and which should be different.

Reply:
Also, can you remind me again why we needed to add `element_repr` to the `QGF` class in the first place? ]

It looks like its a property of the `GF` class that can be easily modified by calling `gf_type.repr(int)`. If this is used mainly as a tool for debugging and visualizing the field elements, then its better to not have it as part of the `QGF` class and simply change the property on the underlying `QGF.galois_field` type wherever needed. 

https://mhostetter.github.io/galois/v0.3.5/api/galois.FieldArray.element_repr/

Reply:
removed